controllers/operator/configmap.go

```diff
@@ -267,7 +268,7 @@ func (r *AuthenticationReconciler) handleConfigMap(instance *operatorv1alpha1.Au
 					currentConfigMap.Data["PREFERRED_LOGIN"] = newConfigMap.Data["PREFERRED_LOGIN"]
 					cmUpdateRequired = true
 				}
-				if _, keyExists := currentConfigMap.Data["MASTER_HOST"]; !keyExists {
+				if _, keyExists := currentConfigMap.Data["MASTER_HOST"]; keyExists {
```

Comment:
This will always overwrite the value and trigger an update on the ConfigMap; I was thinking that you would instead compare the value that you retrieve to the generated one, compare the two, and only set/update when they differ.